### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by oli-obk**
not a great test (shows a bug in clippy xD) but it's a tests nontheless

### CODE REVIEW

**Comment by oli-obk** [comment]
hmm... travis bash is weird...

**Comment by oli-obk** [comment]
ok, this does what it should now :)

File: src/diagnostics.rs [hunk 1]
```diff
@@ -2,27 +2,27 @@
...
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Deserialize, Debug)]
 pub struct DiagnosticSpan {
     pub file_name: String,
-    pub byte_start: usize,
-    pub byte_end: usize,
+    byte_start: u32,
+    byte_end: u32,
```
Comment (killercup) [inline src/diagnostics.rs]: oh noes what if i have _really_ long files? I mean, surely, 4 gigs of rust code should not be a problem, right? :trollface:

File: tests/fixtures.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            } else {
+                let diff = Command::new("git").arg("diff").output().unwrap();
+                File::create("diff.diff").unwrap().write_all(&diff.stdout).unwrap();
+                if !Command::new("diff").arg("-q").arg("diff.diff").arg(test.join("diff.diff")).status().unwrap().success() {
+                    panic!("Unexpected changes applied by rustfix");
+                }
+                if !Command::new("diff").arg("-q").arg("output.txt").arg(test.join("output.txt")).status().unwrap().success() {
+                    panic!("Unexpected output by rustfix");
+                }
+            }
```
Comment (killercup) [inline tests/fixtures.rs]: I love that you went ahead and rewrote that in rust but the whole line of command new things is basically textual noise with a  bunch of "diff" in between ;) Maybe add duct or a crate like that as a dev-dependency? Totally not required right now though

Comment (oli-obk) [inline tests/fixtures.rs]: done

File: src/diagnostics.rs (hunk 1 above)
Comment (oli-obk) [inline src/diagnostics.rs]: I'll leave it to the future generations. That worked with nuclear waste, too.

**Comment by oli-obk** [comment]
> maybe we can just run cargo test with -- --no-capture?

done

File: src/diagnostics.rs (hunk 1 above)
Comment (killercup) [inline src/diagnostics.rs]: That's fair. We are not gonna run rustfix on code generated by lalrpop after all

**Comment by killercup** [comment]
Great work! I'll add some more logging output to the tests in a minute

**Pull request merged**